### PR TITLE
Add one more Include to have the product workable with FetchContent

### PR DESCRIPTION
**Pull request by akabalov**
The issue with FetchContent was opened here https://github.com/oneapi-src/oneMKL/issues/306

The problem is that when we use FetchContent_Declare, it creates ${CMAKE_BINARY_DIR}/_deps dir where we have the build folder, downloaded sources, etc. 

We do have generation of or config.hpp file as a part of oneMKL cmake and it puts this config file into ${CMAKE_CURRENT_BINARY_DIR}/oneapi/mkl/config.hpp folder

So if we don't add paths to the ${CMAKE_CURRENT_BINARY_DIR} into the includes, we end up with failures from domains that this config.hpp is not found. 

To fix the problem, we just need to add ${CMAKE_CURRENT_BINARY_DIR} into includes.

### CODE REVIEW

**Review by mkrainiuk — APPROVED** [review]

**Comment by Rbiessy** [comment]
Thanks for the patch. I can see you are only adding the include path for the mklcpu and mklgpu backends, don't we need this for all the other backends too?
Also just a suggestion I think it would be more readable to store the path to include in `src/CMakeLists.txt` as an internal variable `ONEMKL_GENERATED_INCLUDE_PATH` for instance and use it everywhere instead of relying on relative paths.

**Comment by akabalov** [comment]
> Thanks for the patch. I can see you are only adding the include path for the mklcpu and mklgpu backends, don't we need this for all the other backends too? Also just a suggestion I think it would be more readable to store the path to include in `src/CMakeLists.txt` as an internal variable `ONEMKL_GENERATED_INCLUDE_PATH` for instance and use it everywhere instead of relying on relative paths.

Thank you for pointing me to the missed backends! I love your ideas! Does it look better now?

**Review by Rbiessy — APPROVED** [review]
Thanks, LGTM

File: src/rng/backends/rocrand/CMakeLists.txt [hunk 1]
```diff
@@ -64,7 +64,7 @@ add_library(${LIB_OBJ} OBJECT ${SOURCES})
 
 target_include_directories(
   ${LIB_OBJ} PRIVATE ${PROJECT_SOURCE_DIR}/include ${PROJECT_SOURCE_DIR}/src
-                     ${CMAKE_BINARY_DIR}/bin ${MKL_INCLUDE})
+                     ${CMAKE_BINARY_DIR}/bin ${MKL_INCLUDE} ${ONEMKL_GENERATED_INCLUDE_PATH})
```
Comment (mkrainiuk) [inline src/rng/backends/rocrand/CMakeLists.txt:67]: Could you please check that clang-format won't complain about this line?

Comment (akabalov) [reply src/rng/backends/rocrand/CMakeLists.txt:67]: Checked Clang format using our regular CI job. Everything is fine. Merging changes now.

**Pull request merged**